lib/Doctrine/ORM/Mapping/DefaultQuoteStrategy.php

```diff
@@ -45,9 +45,19 @@ public function getColumnName($fieldName, ClassMetadata $class, AbstractPlatform
      */
     public function getTableName(ClassMetadata $class, AbstractPlatform $platform)
     {
-        return isset($class->table['quoted']) 
-            ? $platform->quoteIdentifier($class->table['name'])
-            : $class->table['name'];
+        $tableName = $class->table['name'];
+
+        if ( ! empty($class->table['schema'])) {
+            $tableName = $class->table['schema'] . '.' . $class->table['name'];
+
+            if ( ! $platform->supportsSchemas() && $platform->canEmulateSchemas()) {
+                $tableName = $class->table['schema'] . '__' . $class->table['name'];
```

Comment:
@Ocramius I am still alive to help on this PR and make it possible to merge it (I didn't see @deeky666 's note before). So it's not a problem if you want me to make some changes!